### PR TITLE
Add test to process key space changes for oauth tokens 

### PR DESCRIPTION
**Pull request by sredxny**
## Description
Added test for the function `ProcessKeySpaceChanges`, it covers the processing of oauth tokens. The test to check the correct processing of ordinary keys will be done in another PR. 

## Motivation and Context
This is part of the work that needs to be done in order to have a testing framework for MDCB

## How This Has Been Tested
`go test -v -run TestProcessKeySpaceChangesForOauth`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: gateway/rpc_storage_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+					}
+					return refresh, err
+				}
+			} else {
+				getKeyFromStore = GlobalSessionManager.Store().GetKey
+				GlobalSessionManager.Store().DeleteAllKeys()
+				GlobalSessionManager.Store().SetKey(token, token, 100)
+				_, err := GlobalSessionManager.Store().GetKey(token)
+				if err != nil {
+					t.Error("Key should be pre-loaded in store previously so the test can perform the revoke action.")
```
Comment (gernest) [inline gateway/rpc_storage_handler_test.go:129]: Why not just `t.Fatal` here? We need the key to be pre loaded I assume, so its safe to just exit here as the rest of the test will generate errors too.

File: go.mod [hunk 2]
```diff
@@ -49,6 +49,7 @@ require (
 	github.com/kelseyhightower/envconfig v1.4.0
 	github.com/lonelycode/go-uuid v0.0.0-20141202165402-ed3ca8a15a93
 	github.com/lonelycode/osin v0.0.0-20160423095202-da239c9dacb6
+	github.com/magiconair/properties v1.8.1
```
Comment (gernest) [inline go.mod:52]: Please run `go mod tidy` I think this was added by accident it is not used anywhere.

File: gateway/rpc_storage_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+func buildStringEvent(eventType, token, apiId string) string {
+	switch eventType {
+	case RevokeOauthHashedToken:
+		// string is as= {the-hashed-token}#hashed:{api-id}:oAuthRevokeToken
+		token = storage.HashStr(token)
+		return fmt.Sprintf("%s#hashed:%s:oAuthRevokeToken", token, apiId)
+	case RevokeOauthToken:
+		// string is as= {the-token}=:{api-id}:oAuthRevokeToken
+		return fmt.Sprintf("%s:%s:oAuthRevokeToken", token, apiId)
```
Comment (gernest) [inline gateway/rpc_storage_handler_test.go:30]: The scheme on the comment is `{the-token}=:{api-id}:oAuthRevokeToken` the format string should be `%s=:%s:oAuthRevokeToken` unless if there is a typo on your comment.

File: gateway/rpc_storage_handler_test.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	case RevokeOauthRefreshToken:
+		// string is as= {the-token}=:{api-id}:oAuthRevokeToken
+		return fmt.Sprintf("%s:%s:oAuthRevokeRefreshToken", token, apiId)
```
Comment (gernest) [inline gateway/rpc_storage_handler_test.go:33]: The scheme on the comment is `{the-token}=:{api-id}: s:oAuthRevokeRefreshToken ` the format string should be `%s=:% s:oAuthRevokeRefreshToken ` unless if there is a typo on your comment.

File: gateway/rpc_storage_handler_test.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	case RevokeOauthRefreshHashedToken:
+		// string is as= {the-token}=:{api-id}:oAuthRevokeToken
+		return fmt.Sprintf("%s:%s:oAuthRevokeToken", token, apiId)
```
Comment (gernest) [inline gateway/rpc_storage_handler_test.go:36]: The scheme on the comment is `{the-token}=:{api-id}: s:oAuthRevokeToken ` the format string should be `%s=:% s:oAuthRevokeToken ` unless if there is a typo on your comment.

File: gateway/rpc_storage_handler_test.go (hunk 3 above)
Comment (sredxny) [reply gateway/rpc_storage_handler_test.go:30]: yes! it was a typo in the comments

File: go.mod (hunk 2 above)
Comment (sredxny) [reply go.mod:52]: done!

File: gateway/rpc_storage_handler_test.go (hunk 1 above)
Comment (sredxny) [reply gateway/rpc_storage_handler_test.go:129]: makes sense, done.

**Comment by sredxny** [comment]
@gernest  thanks for your review, I made the changes :)

**Review by matiasinsaurralde — APPROVED** [review]

**Pull request merged**